requests/adapters.py

```diff
@@ -10,6 +10,10 @@
 
 import os.path
 import socket
+try:
+    from threading import RLock
+except ImportError:  # threading is an optional module and may not be present.
```

Comment:
I don't understand this comment. Also, I don't understand why if this is a problem we've never run into it before given the fact that Urllib3 uses the threading library in several places.